### PR TITLE
Update payment methods to use ids, add address updating

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7200/

**Comment by huwshimi** [comment]
QA:
- open your account page
- delete your payment method (it should get removed)
- add a new payment method (it should get added)
- get your current user and note the addresses you currently have: `app.payment.getUser(app._getUserInfo(app.state).profile, (error, data) => {console.log(error, data)})`
- add a new address: `app.payment.addAddress(app._getUserInfo(app.state).profile, {name: 'address1', line1: '1 Maple St', line2: 'Loonie Lane', state: 'Bunnyhug', city: 'Sasquatch', postcode: '90210', countryCode: 'CA'}, (error) => {console.log(error)})`
- add a new billing address: `app.payment.addBillingAddress(app._getUserInfo(app.state).profile, {name: 'address1', line1: '1 Maple St', line2: 'Loonie Lane', state: 'Bunnyhug', city: 'Sasquatch', postcode: '90210', countryCode: 'CA'}, (error) => {console.log(error)})`
- check that the addresses were added: `app.payment.getUser(app._getUserInfo(app.state).profile, (error, data) => {console.log(error, data)})`
- using the address id returned above update the address: `app.payment.updateAddress(app._getUserInfo(app.state).profile, [address id goes here], {name: 'address2', line1: '9 Kangaroo St', line2: 'In a desert', state: 'Spider Territory', city: 'Snake City', postcode: '9000', countryCode: 'AU'}, (error) => {console.log(error)})`
- using the billing address id returned from the `getUser` call above update your billing address: `app.payment.updateBillingAddress(app._getUserInfo(app.state).profile, [address id goes here], {name: 'address2', line1: '9 Kangaroo St', line2: 'In a desert', state: 'Spider Territory', city: 'Snake City', postcode: '9000', countryCode: 'AU'}, (error) => {console.log(error)})`
- check that the addresses were updated: `app.payment.getUser(app._getUserInfo(app.state).profile, (error, data) => {console.log(error, data)})`

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7201/

**Comment by fabricematrat** [comment]
Just a few things that can be done in another PR:
we need to remove country code, allow entering address when reentering a CC (+checkbox to say same as address)

One small QA issue when having multiple addresses from backend we end up with a big country box in the middle

File: jujugui/static/gui/src/app/jujulib/payment.js [hunk 1]
```diff
@@ -237,22 +244,26 @@ var module = module;
...
           return;
         }
         callback(null, this._parsePaymentMethod(response, true));
       };
-      const id = this._generatePaymentMethodName();
-      const url = `${this.url}/u/${username}/payment-methods/${id}`;
-      return jujulib._makeRequest(this.bakery, url, 'PUT', token, handler);
+      const url = `${this.url}/u/${username}/payment-methods`;
+      const payload = {
+        'payment-method-name': methodName,
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/jujulib/payment.js]: Am I reading right that from `payment-methods.js` will set this 'payment-method-name' to `null`? Is that ok to send in the payload?

Comment (huwshimi) [inline jujugui/static/gui/src/app/jujulib/payment.js]: Yes, that's right, the API doesn't require it and we don't provide a way for the user to enter it.

**Comment by huwshimi** [comment]
Thanks @fabricematrat @Lukewh :ship

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**